src/modules/launcher/Wox.Infrastructure/Storage/StorageHelper.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.IO;
+using System.Windows.Markup;
+
+namespace Wox.Infrastructure.Storage
+{
+    public class StorageHelper
+    {
+        // This detail is accessed by the storage items and is used to decide if the cache must be deleted or not
+        public bool clearCache = false;
+
+        
+        private String currentPowerToysVersion = String.Empty;
+        private String FilePath = String.Empty;
+
+        // As of now this information is not pertinent but may be in the future
+        // There may be cases when we want to delete only the .cache files and not the .json storage files
+        private enum StorageType
```

Comment:
Discussion point:  Instead of BINARY_STORAGE and JSON_STORAGE.  Maybe we want to identify the owner of the version file, and potentially allow for the different storage classes to specify the directory that they are written two.  I'm thinking about the hypothetical case where we allow plugins to be turned off and on, and depending on the user behavior we could end up with caches from different plugins that have different versions of info.   Alternatively if the data itself is versioned than this would not be relevant.

Reply:
When you say the data itself is versioned, do you mean the dataType version number or the powertoys version number in each plugin folder, so that it reads from it's own folder?

There is a pluginJsonStorage which implements the JsonStorage interface. We could use that to make the storage plugin specific.

Reply:
Made the version specific to each storage file.